File: app/src/javaProd/java/com/example/ama/viewer/di/modules/OkHttpModule.java
package com.example.ama.viewer.di.modules;

import com.example.ama.viewer.data.interceptor.RequestHeadersInterceptor;
import com.facebook.stetho.okhttp3.StethoInterceptor;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import okhttp3.OkHttpClient;

@Module
public class OkHttpModule {

    @Provides
    @Singleton
    OkHttpClient provideOkHttpClient(RequestHeadersInterceptor requestHeadersInterceptor,
                                     StethoInterceptor stethoInterceptor) {
        return new OkHttpClient.Builder()
                .addInterceptor(requestHeadersInterceptor)
                .addNetworkInterceptor(stethoInterceptor)
                .build();
    }
}

File: app/src/kotlinDev/java/com/example/ama/viewer/di/modules/OkHttpModule.kt
package com.example.ama.viewer.di.modules

import android.util.Log
import com.example.ama.viewer.data.interceptor.RequestHeadersInterceptor
import com.facebook.stetho.okhttp3.StethoInterceptor
import dagger.Module
import dagger.Provides
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import javax.inject.Singleton

@Module
class OkHttpModule {

    @Provides
    @Singleton
    fun provideOkHttpClient(requestHeadersInterceptor: RequestHeadersInterceptor,
                            stethoInterceptor: StethoInterceptor,
                            loggingInterceptor: HttpLoggingInterceptor): OkHttpClient =
            OkHttpClient.Builder()
                    .addInterceptor(requestHeadersInterceptor)
                    .addInterceptor(loggingInterceptor)
                    .addNetworkInterceptor(stethoInterceptor)
                    .build()

    @Provides
    @Singleton
    fun provideLoggingInterceptor(): HttpLoggingInterceptor =
            HttpLoggingInterceptor { Log.d("LoggingInterceptor", it) }.apply {
                level = HttpLoggingInterceptor.Level.BODY
            }
}